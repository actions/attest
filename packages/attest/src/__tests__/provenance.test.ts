import { generateProvenance } from '../provenance'
import type { Subject } from '../shared.types'

describe('generateProvenance', () => {
  const subject: Subject = {
    name: 'subjecty',
    digest: {
      sha256: '7d070f6b64d9bcc530fe99cc21eaaa4b3c364e0b2d367d7735671fa202a03b32'
    }
  }

  const env = {
    GITHUB_REPOSITORY: 'owner/repo',
    GITHUB_REF: 'refs/heads/main',
    GITHUB_SHA: 'babca52ab0c93ae16539e5923cb0d7403b9a093b',
    GITHUB_WORKFLOW_REF: 'owner/repo/.github/workflows/main.yml@main',
    GITHUB_SERVER_URL: 'https://github.com',
    GITHUB_EVENT_NAME: 'push',
    GITHUB_REPOSITORY_ID: 'repo-id',
    GITHUB_REPOSITORY_OWNER_ID: 'owner-id',
    GITHUB_RUN_ID: 'run-id',
    GITHUB_RUN_ATTEMPT: 'run-attempt',
    RUNNER_ENVIRONMENT: 'github-hosted'
  }

  it('returns a provenance hydrated from env vars', () => {
    const provenance = generateProvenance(subject, env)
    expect(provenance).toMatchSnapshot()
  })
})
