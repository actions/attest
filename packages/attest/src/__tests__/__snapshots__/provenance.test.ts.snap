// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateProvenance returns a provenance hydrated from env vars 1`] = `
{
  "_type": "https://in-toto.io/Statement/v1",
  "predicate": {
    "buildDefinition": {
      "buildType": "https://slsa-framework.github.io/github-actions-buildtypes/workflow/v1",
      "externalParameters": {
        "workflow": {
          "path": ".github/workflows/main.yml",
          "ref": "main",
          "repository": "https://github.com/owner/repo",
        },
      },
      "internalParameters": {
        "github": {
          "event_name": "push",
          "repository_id": "repo-id",
          "repository_owner_id": "owner-id",
        },
      },
      "resolvedDependencies": [
        {
          "digest": {
            "gitCommit": "babca52ab0c93ae16539e5923cb0d7403b9a093b",
          },
          "uri": "git+https://github.com/owner/repo@refs/heads/main",
        },
      ],
    },
    "runDetails": {
      "builder": {
        "id": "https://github.com/actions/runner/github-hosted",
      },
      "metadata": {
        "invocationId": "https://github.com/owner/repo/actions/runs/run-id/attempts/run-attempt",
      },
    },
  },
  "predicateType": "https://slsa.dev/provenance/v1",
  "subject": [
    {
      "digest": {
        "sha256": "7d070f6b64d9bcc530fe99cc21eaaa4b3c364e0b2d367d7735671fa202a03b32",
      },
      "name": "subjecty",
    },
  ],
}
`;
